### ISSUE TITLE
s3.createBucket fails with InvalidLocationConstraint when optional LocationConstraint is not specified

### ISSUE THREAD

**Issue by trivikr** [aws/aws-sdk-js-v3#838]
**Describe the bug**
s3.createBucket fails with InvalidLocationConstraint when optional LocationConstraint is not specified

**SDK version number**
```
$ yarn list aws-sdk
└─ aws-sdk@2.609.0
yarn list @aws-sdk/client-s3
└─ @aws-sdk/client-s3@1.0.0-alpha.8
```

**Is the issue in the browser/Node.js?**
Node.js

**Details of the browser/Node.js version**
v12.14.0

**To Reproduce (observed behavior)**
Verified with the code in https://github.com/trivikr/aws-sdk-js-tests/blob/4441e77012d393f915ded7fce21a06a74ee7f7db/src/node.js with REGION set as `"us-east-1"`

Code:
```javascript
const AWS = require("aws-sdk");
const { S3 } = require("@aws-sdk/client-s3");
const { REGION } = require("./config");

(async () => {
  const bucketName =
    "test-bucket-" +
    Math.random()
      .toString(36)
      .substring(2);
  const params = {
    Bucket: bucketName
  };

  const v2Client = new AWS.S3({ region: REGION });

  console.log(`Creating bucket ${params.Bucket}`);
  await v2Client.createBucket(params).promise();

  console.log(`\nWaiting for "${params.Bucket}" bucket creation`);
  await v2Client.waitFor("bucketExists", params).promise();

  console.log(`\nDeleting bucket ${params.Bucket}`);
  await v2Client.deleteBucket(params).promise();

  params.Bucket = `${bucketName}-v3`;
  const v3Client = new S3({ region: REGION });

  console.log(`\nCreating bucket ${params.Bucket}`);
  await v3Client.createBucket(params);

  console.log(`\nWaiting for "${params.Bucket}" bucket creation`);
  // Use waiters from v3, as v2 waiters are not available
  await v2Client.waitFor("bucketExists", params).promise();

  console.log(`\nDeleting bucket ${params.Bucket}`);
  await v3Client.deleteBucket(params);
})();
```

Output:
```console
$ node src/node.js
Creating bucket test-bucket-8b9tpqq89qq

Waiting for "test-bucket-8b9tpqq89qq" bucket creation

Deleting bucket test-bucket-8b9tpqq89qq

Creating bucket test-bucket-8b9tpqq89qq-v3
(node:13665) UnhandledPromiseRejectionWarning: InvalidLocationConstraint: The specified location-constraint is not valid
    at deserializeAws_restXmlCreateBucketCommandError (/Users/trivikr/workspace/aws-sdk-js-tests/node_modules/@aws-sdk/client-s3/dist/cjs/protocols/Aws_restXml.js:3114:41)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
(node:13665) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 4)
(node:13665) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Expected behavior**
Bucket creation succeeds in v3 with optional LocationConstraint absent

**Comment by AllanFly120** [comment]
This request will fail when region is `us-east-1`. According to [S3 API doc](https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetBucketLocation.html#API_GetBucketLocation_ResponseSyntax), the `location constraint` cannot be `us-east-1`

**Comment by trivikr** [comment]
Confirmed that issue is fixed in `@aws-sdk/client-s3@1.0.0-alpha.10`
Verified with region `us-east-1`

Example output:
```console
$ node src/node.js
Creating bucket test-bucket-kk91lldmjla

Waiting for "test-bucket-kk91lldmjla" bucket creation

Deleting bucket test-bucket-kk91lldmjla

Creating bucket test-bucket-kk91lldmjla-v3

Waiting for "test-bucket-kk91lldmjla-v3" bucket creation

Deleting bucket test-bucket-kk91lldmjla-v3
```
